backend/app/services/scheduling.py

```diff
@@ -0,0 +1,298 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+import shutil
+
+from beanie import PydanticObjectId
+
+from app.config import CHECK_REANA_CONNECTION_INTERVAL, LITTLE_NAP, settings
+from app.helpers import WorkflowState
+from app.models.experiment import Experiment
+from app.models.experiment_run import ExperimentRun
+from app.models.experiment_template import ExperimentTemplate
+from app.routers.aiod import get_dataset_name, get_model_name
+from app.schemas.experiment_run import ExperimentRunId
+from app.schemas.experiment_template import ExperimentTemplateId
+from app.schemas.states import RunState, TemplateState
+from app.services.container_platforms.base_platform import ContainerBasePlatform
+from app.services.workflow_engines.base_engine import (
+    WorkflowBaseEngine,
+    WorkflowConnectionExcpetion,
+)
+
+
+class ExperimentScheduling:
```

Comment:
I recommend rename to `ExperimentScheduler` and module to `experiment_scheduler.py`